integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/CommonUtil.scala

```diff
@@ -288,6 +297,69 @@ object CommonUtil {
     result
   }
 
+  def validateForOverLappingRangeValues(desType: Option[String],
+      rangeInfoArray: Array[String]): Boolean = {
+    val rangeInfoValuesValid = desType match {
+      case Some("IntegerType") | Some("int") =>
+        val intRangeInfoArray = rangeInfoArray.map(_.toInt)
+        val sortedRangeInfoArray = intRangeInfoArray.sorted
+        intRangeInfoArray.sameElements(sortedRangeInfoArray)
+      case Some("StringType") | Some("string") =>
+        val sortedRangeInfoArray = rangeInfoArray.sorted
+        rangeInfoArray.sameElements(sortedRangeInfoArray)
+      case a if (desType.get.startsWith("varchar") || desType.get.startsWith("char")) =>
+        val sortedRangeInfoArray = rangeInfoArray.sorted
+        rangeInfoArray.sameElements(sortedRangeInfoArray)
+      case Some("LongType") | Some("long") | Some("bigint") =>
+        val longRangeInfoArray = rangeInfoArray.map(_.toLong)
+        val sortedRangeInfoArray = longRangeInfoArray.sorted
+        longRangeInfoArray.sameElements(sortedRangeInfoArray)
+      case Some("FloatType") | Some("float") =>
+        val floatRangeInfoArray = rangeInfoArray.map(_.toFloat)
+        val sortedRangeInfoArray = floatRangeInfoArray.sorted
+        floatRangeInfoArray.sameElements(sortedRangeInfoArray)
+      case Some("DoubleType") | Some("double") =>
+        val doubleRangeInfoArray = rangeInfoArray.map(_.toDouble)
+        val sortedRangeInfoArray = doubleRangeInfoArray.sorted
+        doubleRangeInfoArray.sameElements(sortedRangeInfoArray)
+      case Some("ByteType") | Some("tinyint") =>
+        val byteRangeInfoArray = rangeInfoArray.map(_.toByte)
+        val sortedRangeInfoArray = byteRangeInfoArray.sorted
+        byteRangeInfoArray.sameElements(sortedRangeInfoArray)
+      case Some("ShortType") | Some("smallint") =>
+        val shortRangeInfoArray = rangeInfoArray.map(_.toShort)
+        val sortedRangeInfoArray = shortRangeInfoArray.sorted
+        shortRangeInfoArray.sameElements(sortedRangeInfoArray)
+      case Some("BooleanType") | Some("boolean") =>
+        true
+      case a if (desType.get.startsWith("DecimalType") || desType.get.startsWith("decimal")) =>
+        val decimalRangeInfoArray = rangeInfoArray.map(value => BigDecimal(value))
+        val sortedRangeInfoArray = decimalRangeInfoArray.sorted
+        decimalRangeInfoArray.sameElements(sortedRangeInfoArray)
+      case Some("DateType") | Some("date") =>
+        val dateRangeInfoArray = rangeInfoArray.map { value =>
```

Comment:
Dictionary generation can bring duplicate values. duplicate value check required.
Same is the case with timesamp case also.